### PR TITLE
fix(app): make renderer diagnostics passive at idle

### LINKED ISSUE #1516: [Task] Make renderer diagnostics passive at idle
## Goal

A visible, inactive PawWork session performs no animation-frame or periodic sampling solely for renderer diagnostics, while production problem reports retain the passive evidence needed to diagnose long tasks, layout instability, visibility changes, and session lifecycle incidents.

Parent bug: #1515

## Scope

In scope:

- Remove the permanent `requestAnimationFrame` loop, five-second timer, and periodic `renderer.perf.sample` production emission from session performance diagnostics.
- Keep existing structured session, timeline, scroll, refresh, and visibility diagnostics.
- Keep `PerformanceObserver`-based long-task and layout-shift collection, but emit threshold incidents directly from event-driven state rather than a periodic flush.
- Preserve historical diagnostics compatibility in the desktop recorder/sanitizer/problem-report path; old `renderer.perf.sample` JSONL must remain readable.
- Verify the packaged renderer and GPU return to the Electron sleep floor with the watcher disabled only as an experimental isolation variable.

Out of scope:

- Manual performance-recording UI.
- Activity-triggered or bounded automatic RAF sampling.
- macOS file-watcher changes.
- Tauri or other shell migration.
- Removing the renderer diagnostics pipeline.

Accepted tradeoff: automatic problem reports no longer include periodic FPS, frame-gap, jank-count, or heap samples. Existing bounded E2E performance probes remain the frame-performance regression seam. If a future real incident requires compositor-only frame-gap evidence, design a separate explicit profiler from evidence rather than preserving a permanent sampler now.

## Relevant files or context

- `packages/app/src/context/renderer-diagnostics.ts`
- `packages/app/src/context/renderer-diagnostics.test.ts`
- `packages/app/src/pages/session/use-session-page-diagnostics.ts`
- `packages/app/e2e/session/session-renderer-diagnostics.spec.ts`
- `packages/app/e2e/perf/probe.ts`
- `packages/desktop-electron/src/main/renderer-diagnostics-*`
- Parent diagnosis and measured baselines: #1515

Historical context:

- #389 / #392 introduced the incident evidence pipeline.
- #432 and #647 used long-task, CLS, and lifecycle evidence to identify root causes; FPS/frame-gap data was supporting evidence, not the unique causal signal.

## Verification

- Add focused tests proving session-page mount, visible idle, and visibility changes schedule zero RAF callbacks and zero periodic timers for diagnostics.
- Simulated long tasks at the existing incident threshold emit the expected jank incident without polling.
- Layout-shift entries use an event-driven session-window calculation and emit when the existing CLS threshold is crossed.
- Cleanup prevents all later emissions.
- Existing renderer diagnostics E2E coverage remains green.
- Run the relevant app tests, typecheck, and desktop-host verification required by `AGENTS.md`.
- Build/run the real desktop app. With the file watcher disabled only to isolate this slice, run five frozen idle repetitions and report renderer/GPU CPU and context-switch results against the #1515 baseline.
- Do not remove old `renderer.perf.sample` from desktop allowlists or parsing; verify historical diagnostic lines still parse.

## Execution mode

Implement and open a PR — create a flat branch from current `dev`, implement this slice only, run targeted verification, push, and open a PR following the repository template. Link both this task and #1515. Do not wait for the watcher slice and do not stack on it.

### PR DESCRIPTION
**Pull request by Astro-Han**
## Summary

- Remove the session diagnostics RAF loop, five-second interval, and production `renderer.perf.sample` emission.
- Emit long-task and CLS threshold incidents directly from `PerformanceObserver` callbacks, with CLS session-window accounting and cleanup guards.
- Keep visibility and existing session/timeline/scroll/refresh diagnostics intact, and lock historical `renderer.perf.sample` JSONL parsing with a desktop regression test.

## Why

Visible idle session pages continuously scheduled animation frames and periodic diagnostics samples. That diagnostic-only activity kept the renderer and GPU awake even when the user and session were inactive.

## Related Issue

Closes #1516

Parent context: #1515 (this PR does not close the parent issue).

## Human Review Status

Pending

## Review Focus

Please focus on the event-driven long-task/CLS thresholds, the CLS session-window boundaries, and the guarantee that observer callbacks cannot emit after cleanup.

## Risk Notes

- Automatic problem reports intentionally stop receiving periodic FPS, frame-gap, jank-count, and heap samples. Historical `renderer.perf.sample` JSONL remains readable.
- The diagnostics E2E file has two unrelated `dev` baseline failures: the scroll-reason assertion and worktree entry timeout fail identically from an isolated `origin/dev` worktree. The other two scenarios pass.
- No visible UI or copy changed, so manual screenshots/recordings are not applicable.
- No platform or packaging implementation changed; a real arm64 macOS packaged build and smoke run were still verified.
- No docs, release notes, dependencies, permissions, credentials, deletion behavior, or generated content changed.

## How To Verify

```text
Focused renderer diagnostics tests: 13 passed (RED/GREEN coverage for zero RAF/timers, event-driven long-task/CLS incidents, and cleanup)
Historical desktop diagnostics compatibility: 8 passed, including old renderer.perf.sample JSONL parsing
App unit suite: 2016 passed
Desktop host unit suite: 641 passed
App typecheck: passed
Desktop typecheck and release typecheck: passed
Changed-source ESLint and Prettier: passed (test files are excluded by the repository ESLint config)
Diagnostics E2E: 2 passed; 2 failed identically on origin/dev baseline (documented in Risk Notes)
Desktop production build: passed
Packaged arm64 macOS smoke with a real renderer CDP target: passed

Five fresh packaged visible-session idle runs, watcher disabled only as an isolation variable:
run 1 — renderer 0.013% CPU / 4.069 csw/s; GPU 0.000% CPU / 1.138 csw/s
run 2 — renderer 0.313% CPU / 15.172 csw/s; GPU 0.000% CPU / 0.828 csw/s
run 3 — renderer 0.327% CPU / 18.172 csw/s; GPU 0.040% CPU / 9.414 csw/s
run 4 — renderer 0.313% CPU / 16.414 csw/s; GPU 0.087% CPU / 17.793 csw/s
run 5 — renderer 0.240% CPU / 15.655 csw/s; GPU 0.000% CPU / 2.000 csw/s
mean  — renderer 0.241% CPU / 13.896 csw/s; GPU 0.025% CPU / 6.235 csw/s

Protocol: isolated fresh user data, real packaged session route confirmed by session.timeline.mount JSONL, 20-second settle, 31 one-second top samples, first sample discarded. No renderer.perf.sample events were produced.
```

## Screenshots or Recordings

Not applicable; there is no visible UI change.

## Checklist

> **How to use this checklist:**
>
> - Tick a box by replacing `[ ]` with `[x]`. Do not edit, add, or remove items.
> - The bot-applied label items can only be honestly ticked AFTER the PR is opened and the labeler / priority-triage bots have run — return to the PR description and tick them then.
> - Most items are required. The few that are conditional are explicitly marked **(conditional)**; for those, leave unticked if they truly do not apply and explain why in Risk Notes. All other items must be ticked before requesting human review.

- [x] **Type label** — this PR carries exactly one of `bug`, `enhancement`, `task`, `documentation`. Type labels are author-added; the labeler bot does NOT assign them. Add the label in the GitHub UI, then tick this.
- [x] **Routing labels** — this PR carries at least one of `app`, `ui`, `platform`, `harness`, `ci`. The labeler bot assigns these on PR open based on changed paths. Confirm the bot's choice (or override if wrong), then tick this.
- [x] **Priority label** — this PR carries exactly one of `P0`, `P1`, `P2`, `P3`. The priority-triage bot suggests one on PR open. Confirm or override, then tick this.
- [x] Human Review Status above is set to `Pending`, `Approved by @<reviewer>`, or `Not required: <reason>` (default is `Pending`; "not required" is restricted to bot-authored low-risk PRs).
- [x] I linked the related issue, or stated in Summary why there is no issue.
- [x] I described the review focus and any meaningful risks.
- [x] I replaced the example block in How To Verify with the real verification steps and the key result for each.
- [x] I did not introduce unrelated refactors, dependencies, generated files, or file changes beyond the stated scope.
- [ ] **(conditional)** I manually checked visible UI or copy changes when needed, with screenshots or recordings. Leave unticked only if no visible UI or copy changed.
- [x] **(conditional)** I considered macOS and Windows impact for platform, packaging, updater, signing, paths, shell, or permissions changes. Leave unticked only if no platform/packaging surface was touched.
- [ ] **(conditional)** I called out docs, release notes, dependencies, permissions, credentials, deletion behavior, generated content, or local file changes when relevant. Leave unticked only if none of those surfaces was touched.
- [x] I reviewed the final diff for unrelated changes and suspicious dependency changes.
- [x] I am targeting `dev`, and my PR title and commit messages use Conventional Commits in English.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved session performance diagnostics to detect long-task and layout-shift incidents more reliably.
  * Prevented duplicate or stale diagnostic events after a session ends or its interface is disposed.
  * Reduced unnecessary performance monitoring while sessions are visible but idle.
  * Preserved renderer performance data when diagnostic events are parsed and sanitized.

* **Tests**
  * Expanded coverage for performance incidents, layout shifts, session cleanup, and diagnostic event parsing.

### CODE REVIEW

**Comment by Astro-Han** [comment]
@coderabbitai review

File: packages/desktop-electron/src/main/renderer-diagnostics-sanitize.test.ts [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
+      data: {
+        fps: 60,
+        frame_gap_ms: 17,
+        jank_count: 1,
+        long_task_max_ms: 120,
+        long_task_block_ms: 140,
+        cls: 0.02,
+        heap_used_mb: 256,
+      },
+    })
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/desktop-electron/src/main/renderer-diagnostics-sanitize.test.ts:40]: Addressed in 7495c0d063: the historical renderer.perf.sample JSONL regression test now asserts both app_launch_id and window_id. The focused sanitizer tests (8/8), full desktop suite (641/641), and desktop typecheck pass.
